src/mutation.cpp

```diff
@@ -308,66 +291,65 @@ void Character::mutation_loss_effect(std::string mut)
 
 bool Character::has_active_mutation(const std::string & b) const
 {
-    const auto &mut_iter = my_mutations.find( b );
-    if( mut_iter == my_mutations.end() ) {
-        return false;
-    }
-    return traits[*mut_iter].powered;
+    const auto iter = my_mutations.find( b );
+    return iter != my_mutations.end() && iter->second.powered;
 }
 
-void player::activate_mutation( std::string mut )
+void player::activate_mutation( const std::string &mut )
 {
-    int cost = traits[mut].cost;
+    const auto &mdata = mutation_branch::get( mut );
+    auto &tdata = my_mutations[mut];
+    int cost = mdata.cost;
     // You can take yourself halfway to Near Death levels of hunger/thirst.
     // Fatigue can go to Exhausted.
-    if ((traits[mut].hunger && hunger >= 700) || (traits[mut].thirst && thirst >= 260) ||
-      (traits[mut].fatigue && fatigue >= 575)) {
+    if ((mdata.hunger && hunger >= 700) || (mdata.thirst && thirst >= 260) ||
+      (mdata.fatigue && fatigue >= 575)) {
       // Insufficient Foo to *maintain* operation is handled in player::suffer
-        add_msg(m_warning, _("You feel like using your %s would kill you!"), traits[mut].name.c_str());
+        add_msg(m_warning, _("You feel like using your %s would kill you!"), mdata.name.c_str());
         return;
     }
-    if (traits[mut].powered && traits[mut].charge > 0) {
+    if (tdata.powered && tdata.charge > 0) {
         // Already-on units just lose a bit of charge
-        traits[mut].charge--;
+        tdata.charge--;
```

Comment:
Yeah, here's charge.  It's supposed to handle timing the active traits that take X resources per T time.